src/qibolab/instruments/zhinst.py

```diff
@@ -872,109 +872,122 @@ def play_sweep(self, exp, qubit, pulse, section):
                 )
 
-    def couplerflux(self, exp, couplers):
-        """Coupler flux for bias sweep or pulses."""
+    def couplerflux(self, exp: lo.Experiment, couplers: Dict[str, Coupler]):
+        """Coupler flux for bias sweep or pulses.
+
+        Args:
+            exp (lo.Experiment): laboneq experiment on which register sequences.
+            couplers (dict[str, Coupler]): coupler on which pulses are played.
+        """
         for coupler in couplers.values():
             c = coupler.name  # pylint: disable=C0103
-            with exp.section(uid=f"sequence_bias_coupler{c}"):
-                i = 0
-                time = 0
-                for pulse in self.sequence[f"couplerflux{c}"]:
-                    pulse.zhpulse.uid += str(i)
-                    exp.delay(
-                        signal=f"couplerflux{c}",
-                        time=round(pulse.pulse.start * NANO_TO_SECONDS, 9) - time,
-                    )
-                    time = round(pulse.pulse.duration * NANO_TO_SECONDS, 9) + round(
-                        pulse.pulse.start * NANO_TO_SECONDS, 9
-                    )
-                    if isinstance(pulse, ZhSweeperLine):
-                        self.play_sweep(exp, coupler, pulse, section="couplerflux")
-                    elif isinstance(pulse, ZhSweeper):
-                        self.play_sweep(exp, coupler, pulse, section="couplerflux")
-                    elif isinstance(pulse, ZhPulse):
-                        exp.play(signal=f"couplerflux{c}", pulse=pulse.zhpulse)
-                    i += 1
-
-    def flux(self, exp, qubits):
-        """Qubit flux for bias sweep or pulses."""
+            time = 0
+            i = 0
+            previous_section = None
+            for j, sequence in enumerate(self.sub_sequences[f"couplerflux{c}"]):
+                section_uid = f"sequence_couplerflux{c}_{j}"
+                with exp.section(uid=section_uid, play_after=previous_section):
+                    for pulse in sequence:
+                        pulse.zhpulse.uid += str(i)
+                        exp.delay(
+                            signal=f"couplerflux{c}",
+                            time=round(pulse.pulse.start * NANO_TO_SECONDS, 9) - time,
+                        )
+                        time = round(pulse.pulse.duration * NANO_TO_SECONDS, 9) + round(
+                            pulse.pulse.start * NANO_TO_SECONDS, 9
+                        )
+                        if isinstance(pulse, ZhSweeperLine):
+                            self.play_sweep(exp, coupler, pulse, section="couplerflux")
+                        elif isinstance(pulse, ZhSweeper):
+                            self.play_sweep(exp, coupler, pulse, section="couplerflux")
+                        elif isinstance(pulse, ZhPulse):
+                            exp.play(signal=f"couplerflux{c}", pulse=pulse.zhpulse)
+                        i += 1
```

Comment:
```suggestion
            previous_section = None
            for i, sequence in enumerate(self.sub_sequences[f"couplerflux{c}"]):
                section_uid = f"sequence_couplerflux{c}_{i}"
                with exp.section(uid=section_uid, play_after=previous_section):
                    for j, pulse in enumerate(sequence):
                        pulse.zhpulse.uid += f"{i}_{j}"
                        exp.delay(
                            signal=f"couplerflux{c}",
                            time=round(pulse.pulse.start * NANO_TO_SECONDS, 9) - time,
                        )
                        time = round(pulse.pulse.duration * NANO_TO_SECONDS, 9) + round(
                            pulse.pulse.start * NANO_TO_SECONDS, 9
                        )
                        if isinstance(pulse, ZhSweeperLine):
                            self.play_sweep(exp, coupler, pulse, section="couplerflux")
                        elif isinstance(pulse, ZhSweeper):
                            self.play_sweep(exp, coupler, pulse, section="couplerflux")
                        elif isinstance(pulse, ZhPulse):
                            exp.play(signal=f"couplerflux{c}", pulse=pulse.zhpulse)
```


Can we do this, or does `zhpulse.uid` need to be exactly what you have? If not I suggest this change to improve the readability a bit more.

Reply:
I don't see any issues with this change. I'll do it also for `drive` and  `flux`.